kedro-datasets/kedro_datasets/pandas/parquet_dataset.py

```diff
@@ -214,3 +216,19 @@ def _invalidate_cache(self) -> None:
         """Invalidate underlying filesystem caches."""
         filepath = get_filepath_str(self._filepath, self._protocol)
         self._fs.invalidate_cache(filepath)
+
+    def preview(self, nrows: int = 5) -> TablePreview:
+        """
+        Generate a preview of the dataset with a specified number of rows.
+
+        Args:
+            nrows: The number of rows to include in the preview. Defaults to 5.
+
+        Returns:
+            dict: A dictionary containing the data in a split format.
+        """
+        load_path = str(self._get_load_path())
+        data_preview = pd.read_parquet(load_path, **self._load_args)
+        preview_data = data_preview.head(nrows).to_dict(orient="split")
```

Comment:
Does this read the full file and then filter? This will cause significant performance issue, there should be native parquet options that allows you to filter from source.

Reply:
Just going through [pandas.read_parquet](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_parquet.html) I don't see an option to read in a specific number of rows and the filter parameter seems to be filtering the data based on column values and not the number of rows which isn't super helpful as this is a generic method.

Maybe we should go with a lower level approach and import `pyarrow`?